trying to add indexOf to array (#1352)
onflow/cadence: docs/language/values-and-types.mdx

  let containsKitty = numbers.contains("Kitty")
  ```

> - `cadence•fun indexOf(_ element: T): Int?`

Reviewer: I'd recommend renaming this function to `firstIndex(of: T): Int?`, [like in Swift](https://developer.apple.com/documentation/swift/array/2994720-firstindex)

Author: good point, will do.

 Returns the index of the first matching elmement of type `T`, or nil if no match

 ```cadence
  // Declare an array of integers.
  let numbers = [42, 23, 31, 12]

  //Check if the array contains 31

>   let index= numbers.indexOf(31)
>   // `index` is 2
> 
>   //Check if the array contains 22
>   let index= numbers.indexOf(31)
>   // `index` is nil

Reviewer: ```suggestion
  let index = numbers.indexOf(31)
  // `index` is 2

  //Check if the array contains 22
  let index = numbers.indexOf(31)
  // `index` is nil
```

  let containsKitty = numbers.contains("Kitty")
  ```

> - `cadence•fun firstIndex(of:: T): Int?`
>  Returns the index of the first matching elmement of type `T`, or nil if no match

Reviewer: ```suggestion
- `cadence•fun firstIndex(of: T): Int?`
 Returns the index of the first matching element of type `T`, or nil if no match.
 Available if `T` is not resource-kinded and equatable.
```

 ```cadence
  // Declare an array of integers.
  let numbers = [42, 23, 31, 12]

>   //Check if the array contains 31
> 
>   let index= numbers.firstIndex(of:31)
>   // `index` is 2
> 
>   //Check if the array contains 22
>   let index= numbers.firstIndex(of:22)
>   // `index` is nil

Reviewer: ```suggestion
  // Check if the array contains 31
  let index = numbers.firstIndex(of: 31)
  // `index` is 2

  // Check if the array contains 22
  let index = numbers.firstIndex(of: 22)
  // `index` is nil
```
